saleor/graphql/order/mutations/orders.py

```diff
@@ -19,6 +19,41 @@ def try_payment_action(action, money, errors):
         errors.append(Error(field='payment', message=str(e)))
 
 
+def clean_order_cancel(order, errors):
+    if order and not order.can_cancel():
+        errors.append(
+            Error(
+                field='order',
+                message='This order can\'t be canceled.'))
+    return errors
+
+
+def clean_order_mark_as_paid(order, errors):
+    if order and order.payments.exists():
+        errors.append(
+            Error(
+                field='payment',
+                message='Orders with payments can not be manually '
+                        'marked as paid.'))
+    return errors
+
+
+def clean_order_capture(payment, amount, errors):
+    if not payment:
+        errors.append(
+            Error(
+                field='payment',
+                message='There\'s no payment associated with the order.'))
+        return errors
+    if payment.status != PaymentStatus.PREAUTH:
+        errors.append(
+            Error(
+                field='payment',
+                message='Only pre-authorized payments can be captured'))
+    try_payment_action(payment.capture, amount, errors)
```

Comment:
I see some inconsistency here - rest of those `clean_*` functions only check if action preconditions are true and return errors respectively, while this one as well tries to perform an action. Is that on purpose? I'd expect only logical checks here.